pkgs/os-specific/linux/kmscon/default.nix

```
@@ -33,14 +36,22 @@ stdenv.mkDerivation rec {
     pango
     pixman
     systemd
+    mesa
+    ninja
   ];
 
   nativeBuildInputs = [
-    autoreconfHook
+    meson
     docbook_xsl
     pkg-config
   ];
 
+  patches = [ ./0001-tests-fix-warnings.patch ]; # https://github.com/Aetf/kmscon/commit/b65f4269b03de580923ab390bde795e7956b633f
```

Put the comments before the patches. It is more readable.
```
  patches = [ 
    # https://github.com/Aetf/kmscon/commit/b65f4269b03de580923ab390bde795e7956b633f
    ./0001-tests-fix-warnings.patch 
  ];
```
----------

Way better, in this case you should use fetchpatch

```suggestion
  patches = [
    (fetchpatch {
       name = "0001-tests-fix-warnings.patch";
       # Below is a trick: add `.patch` extension after the GitHub URL, and it will show you a raw patch file
       url = "https://github.com/Aetf/kmscon/commit/b65f4269b03de580923ab390bde795e7956b633f.patch";
       sha256 = ""; # run the expression, and the error message will tell you the true, expected hash 
     };)
  ];

---

You know, I remember trying to get the `.patch` of a commit like that via the URL and it just didn't work, I guess I didn't quite use the right URL. That is definitely the better method, updating the PR now.